### ISSUE TITLE
Wrong rolearn in aws_auth_configmap_yaml

### ISSUE THREAD

**Issue by Anorlondo448** [terraform-aws-modules/terraform-aws-eks#1819]
## Description

"data.maproles[].rolearn" set in aws_auth_configmap_yaml is not “Pod Execution Role ARN”. It is “Fargate Profile ARN”.
The correct policy is not attached to the user mapped in the aws-auth ConfigMap.

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/16f46db94b7158fd762d9133119206aaa7cf6d63/templates/aws_auth_cm.tpl#L30-L37

## Versions
 - Terraform: v1.1.4
 - Provider(s):
```
+ provider registry.terraform.io/hashicorp/aws v3.74.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/kubernetes v1.13.4
+ provider registry.terraform.io/hashicorp/tls v3.1.0
```

## Reproduction
- use `terraform-aws-modules/eks/aws`
- Use only fargate_profile, not nodegroups.
- output `module.eks.aws_auth_configmap_yaml`

### Code Snippet to Reproduce
- output value
```
Changes to Outputs:
  + aws_auth_configmap_yaml = <<-EOT
        apiVersion: v1
        kind: ConfigMap
        metadata:
          name: aws-auth
          namespace: kube-system
        data:
          mapRoles: |
            - rolearn: arn:aws:eks:<region>:<AWS account id>:fargateprofile/<cluster name>/<fargate profile name>/xxxxxxxx
              username: system:node:{{SessionName}}
              groups:
                - system:bootstrappers
                - system:nodes
                - system:node-proxier
    EOT
```

## Expected behavior
"data.maproles[].rolearn" should be set to “Pod Execution Role ARN”.

## Actual behavior
set to "Fargate Profile ARN"

## Additional context
- The output of aws_auth_configmap_yaml does not have a Pod Execution Role ARN
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/be86c0b898c34943e898e2ecd4994bb7904663ff/outputs.tf#L164-L172

- templates/aws_auth_cm.tpl should also use the "Pod Execution Role ARN", not "Fargate Profile ARN"
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/16f46db94b7158fd762d9133119206aaa7cf6d63/templates/aws_auth_cm.tpl#L30-L32

- The examples are also Fargate Profile ARN
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c45adce32e4a109f18c7069b5415e91679deab16/examples/complete/main.tf#L299-L319

**Comment by bryantbiggs** [comment]
I remember having this specified as you have in your PR, but I believe with testing, it was determined that the profile ARN is what is necessary in the config-map. 

I don't have time at the moment, but one way to know for certain is to deploy the Fargate profile example https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/fargate_profile and verify what IAM resource is added in the config-map. When using EKS managed node groups or Fargate profiles, AWS updates the config-map with the associated roles. The output attribute provided is intended to mimic that functionality. I remember being shocked that the profile ARN was in the config map and not the IAM role - but its been a bit since then and I could have missed something

**Comment by Anorlondo448** [comment]
thanks @bryantbiggs !

I actually checked by outputting aws_auth_configmap_yaml, but what was set was the Fargate Profile ARN.

```
Changes to Outputs:
  + aws_auth_configmap_yaml = <<-EOT
        apiVersion: v1
        kind: ConfigMap
        metadata:
          name: aws-auth
          namespace: kube-system
        data:
          mapRoles: |
            - rolearn: arn:aws:eks:<region>:<AWS account id>:fargateprofile/<cluster name>/<fargate profile name>/xxxxxxxx
              username: system:node:{{SessionName}}
              groups:
                - system:bootstrappers
                - system:nodes
                - system:node-proxier
    EOT
```

The ConfigMap should have the ARN of the IAM role, but you will not get the correct IAM policy even if the Fargate Profile ARN is specified.

**Comment by Anorlondo448** [comment]
I created a PR that sets the Pod Execution Role Arn, not the Fargate Profile ARN.
https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1820

**Comment by bryantbiggs** [comment]
ah yes, just re-ran this morning and you are correct. your PR looks good as well if you can just run the pre-commit hooks on it

**Comment by bryantbiggs** [comment]
closed in #1820

**Comment by Anorlondo448** [comment]
@bryantbiggs 
thanks!
